tests/testConstantSettings.php

```diff
@@ -88,19 +88,22 @@ public function testThatConstantValuesAreUsed() {
 	 * @runInSeparateProcess
 	 */
 	public function testSetWithConstant() {
-		$opt_name     = 'domain';
 		$constant_val = rand();
 
 		// Set a constant and make sure it works.
 		define( self::CONSTANT_PREFIX . 'DOMAIN', $constant_val );
 		$opts = new WP_Auth0_Options();
-		$this->assertEquals( $constant_val, $opts->get( $opt_name ) );
+		$this->assertEquals( $constant_val, $opts->get( 'domain' ) );
 
 		// Try to set an option with the constant set.
 		$new_value  = str_shuffle( $constant_val );
-		$set_result = $opts->set( $opt_name, $new_value, false );
+		$set_result = $opts->set( 'domain', $new_value );
 		$this->assertFalse( $set_result );
-		$this->assertEquals( $constant_val, $opts->get( $opt_name ) );
+		$this->assertEquals( $constant_val, $opts->get( 'domain' ) );
+
+		// Test that the constant value remains if a new option is set.
+		$opts->set( 'client_id', $new_value );
```

Comment:
This the is test for the issue found while testing updates.

Reply:
I don't understand what's being tested here. The steps I can follow are:
- `$constant_val` is a random string.
- You define a constant named "DOMAIN" with that value.
- You verify the value matches what you get via `$opts->get("domain")`. (How can I know that `$opts->get( 'domain' )` is going to take the value from the defined constant? )

- you then try to overwrite the "domain" option and fail. 
- You assert the value matches what was set initially. Still the value from the constant 👍 .
- But then you try to set client_id to some value, and still right below check again that "domain" option is still the constant. So:

1) Why checking twice that the `$opts->get( 'domain' )` is equal to the constant defined?
2) Why setting `client_id` option if not being asserted below?

Reply:
Yes, a little confusing 😄 which is what made it take so long to figure out where the 🐞 was coming from.

The constant settings are not stored in the database so, before saving, they are removed from the array that's saving it. The 🐞 here was that these were being removed from memory as well (what `get()` accesses) instead of only from the ones saved to the database. So changing a different option and then saving would change the correct option and save the correct array but remove all constant-stored settings for the rest of the request. On the next page load, all would be fine (unless the save happens again).